ui/gtk4.c

```diff
@@ -0,0 +1,2176 @@
+    gtk_application_set_accels_for_action(app, "win.show-menubar",
+                                          show_menubar_accels);
+    gtk_application_set_accels_for_action(app, "win.fullscreen",
+                                          fullscreen_accels);
+    gtk_application_set_accels_for_action(app, "win.quit",
+                                          quit_accels);
+}
+
+static void on_app_activate(GtkApplication *app, GtkDisplayState *ds)
+{
+    VirtualConsole *vc;
+    GtkIconTheme *theme;
+    GMenu *menu;
+    GdkDisplay *display;
+    char *dir;
+
+    g_set_prgname("qemu");
+
+
+    ds->window = gtk_application_window_new(app);
+    gtk_application_window_set_show_menubar(GTK_APPLICATION_WINDOW(ds->window),
+                                            TRUE);
+    ds->vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 0);
+    ds->notebook = gtk_notebook_new();
+    menu = gd_create_menus_models(ds, ds->opts);
+    gtk_application_set_menubar(ds->app, G_MENU_MODEL(menu));
+    setup_actions(ds, GTK_APPLICATION_WINDOW(ds->window));
+
+    display = gtk_widget_get_display(ds->window);
+    theme = gtk_icon_theme_get_for_display(display);
+    dir = get_relocated_path(CONFIG_QEMU_ICONDIR);
+    gtk_icon_theme_add_search_path(theme, dir);
+    g_free(dir);
+
+    ds->free_scale = FALSE;
+
+    /*
+     * Mostly LC_MESSAGES only. See early_gtk_display_init() for details. For
+     * LC_CTYPE, we need to make sure that non-ASCII characters are considered
+     * printable, but without changing any of the character classes to make
+     * sure that we don't accidentally break implicit assumptions.
+     */
+    setlocale(LC_MESSAGES, "");
+    setlocale(LC_CTYPE, "C.UTF-8");
+    dir = get_relocated_path(CONFIG_QEMU_LOCALEDIR);
+    bindtextdomain("qemu", dir);
+    g_free(dir);
+    bind_textdomain_codeset("qemu", "UTF-8");
+    textdomain("qemu");
+
+    if (ds->opts->has_show_cursor && ds->opts->show_cursor) {
+        ds->null_cursor = NULL; /* default pointer */
+    } else {
+        ds->null_cursor = gdk_cursor_new_from_name("none", NULL);
+    }
+
+    ds->mouse_mode_notifier.notify = gd_mouse_mode_change;
+    qemu_add_mouse_mode_change_notifier(&ds->mouse_mode_notifier);
+    qemu_add_vm_change_state_handler(gd_change_runstate, ds);
+
+    gtk_window_set_icon_name(GTK_WINDOW(ds->window), "qemu");
+
+    gd_connect_signals(ds);
+
+    gtk_notebook_set_show_tabs(GTK_NOTEBOOK(ds->notebook), FALSE);
+    gtk_notebook_set_show_border(GTK_NOTEBOOK(ds->notebook), FALSE);
+
+    gd_update_caption(ds);
+
+    gtk_widget_set_vexpand(ds->vbox, TRUE);
+    gtk_box_append(GTK_BOX(ds->vbox), ds->notebook);
+
+    gtk_window_set_child(GTK_WINDOW(ds->window), ds->vbox);
+
+    gtk_window_present(GTK_WINDOW(ds->window));
+    if (ds->opts->u.gtk.has_show_menubar &&
+        !ds->opts->u.gtk.show_menubar) {
+        gtk_application_window_set_show_menubar(
+            GTK_APPLICATION_WINDOW(ds->window),
+            FALSE
+        );
```

Comment:
The end of the statement should be on the previous line as you do in all the other occurrences where you split it.

Reply:
Not sure I understand this comment, would you mind explaining what you meant?

Reply:
I meant that the `);` should not be in a stand-alone line, this is the only place where I saw it coded that way.
```c
        gtk_application_window_set_show_menubar(
            GTK_APPLICATION_WINDOW(ds->window),
            FALSE);
```